### ISSUE TITLE
Return default location in for azure region lookup

### ISSUE THREAD

**Issue by hugorut** [infracost/infracost#1824]
The azure `lookupRegion` function fails to return a default location if the location cannot be fetched from an attribute. e.g:

```
resource "azurerm_mssql_server" "main" {
  name                         = "bousql-${var.environment}"
  resource_group_name          = data.azurerm_resource_group.main.name
  location                     = data.azurerm_resource_group.main.location
  ...
}

resource "azurerm_mssql_database" "main" {
  sku_name = local.env-test == local.env-acc ? "S3" : "S1"
  ...
}
```

In the above example HCL will fail to return a `location` for `azurerm_mssql_server.main` and thus `azurerm_mssql_database.main` will use a blank location. In these instances, we should return a sane default. Then when https://github.com/infracost/infracost/issues/1796 is implemented users will be able to configure this region manually.

**Comment by aliscott** [comment]
Is this a bug with `DefaultProviderRegion` ([link](https://github.com/infracost/infracost/blob/master/internal/providers/terraform/azure/azure.go#L8)) not being used properly?

**Comment by hugorut** [comment]
Yeah it must be, I think it's most likely because of incorrect empty string check somewhere

**Comment by hugorut** [comment]
This has been fixed and is now in master awaiting the next release.

**Comment by hugorut** [comment]
Changes addressing this issue have been published in the [v0.10.7](https://github.com/infracost/infracost/releases/tag/v0.10.7) release.
